src/sql/src/plan/statement.rs

```diff
@@ -1115,3 +1122,80 @@ impl<T: mz_sql_parser::ast::AstInfo> From<&Statement<T>> for StatementClassifica
         }
     }
 }
+
+/// Validates this statement can be planned in the provided context.
+fn validate_statement<'a>(
+    stmt: &'a Statement<Aug>,
+    ctx: &'a StatementContext<'a>,
+) -> Result<(), PlanError> {
+    let mut validator = StatementValidator::new(ctx);
+    // Recursively visits the different parts of the Statement.
+    validator.visit_statement(stmt);
+
+    // Statement validation has the possibility of causing sticky panics if
+    // something fails when openning the catalog, so we give ourselves an out.
+    if SKIP_STATEMENT_VALIDATION.get(ctx.catalog.system_vars().dyncfgs()) {
+        if validator.state.is_err() {
+            tracing::warn!(
+                error = ?validator.state,
+                "skipping statement validation when there was an error"
+            );
+        }
+        validator.state = Ok(());
+    }
+
+    // Return the state the validator was left in.
+    validator.state
+}
```

Comment:
FWIW, the way that we've handled this in the past is to due the validation in sequencing (for example in `sequence_create_table`). Generally an item only goes through sequencing when it's first created so we know that all user typed DDL will go through sequencing, but other DDL will not.

Here's a recent example: https://github.com/MaterializeInc/materialize/blob/8d1128a98d1887a336d59b65d6238a755276796e/src/adapter/src/coord/sequencer/inner.rs#L660-L678

Reply:
I'm not necessarily saying you should remove this validation, but I'm just wondering if there's a good reason to prefer doing it in planning instead of sequencing?

Reply:
Originally I thought a specific validation step in planning would be more clear, but I think putting it in sequencing is easier and fits with the status quo so I removed the new validation logic and will add the check in sequencing in a follow up PR.

Note: It's not in this PR yet because the `VERSION` constraint gets rejected in planning as unsupported, I updated the SLT test to reflect that